### PR TITLE
fix: Appends annotations returned by plugin to signature manifest's annotations

### PR DESCRIPTION
**Pull request by priteshbandi**
Another way to implement https://github.com/notaryproject/notation-go/pull/253 as per suggestion from @patrickzheng200 

-------
As per spec([link](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#generate-envelope)) annotations returned by plugin should be appened to signature manifest's annotations and this PR introduces aforementioned behavior

Signed-off-by: Pritesh Bandi <pritesb@amazon.com>

### CODE REVIEW

File: signer/plugin.go [hunk 1]
```diff
@@ -21,12 +21,13 @@ import (
 // pluginSigner signs artifacts and generates signatures.
 // It implements notation.Signer
 type pluginSigner struct {
-	plugin       plugin.SignPlugin
-	keyID        string
-	pluginConfig map[string]string
+	plugin              plugin.SignPlugin
+	keyID               string
+	pluginConfig        map[string]string
+	manifestAnnotations map[string]string
```
Comment (patrickzheng200) [inline signer/plugin.go:27]: nit: can we rename it to `pluginAnnotations`? since it's annotations from the plugin.

File: signer/plugin.go [hunk 2]
```diff
@@ -44,6 +45,10 @@ func NewFromPlugin(plugin plugin.Plugin, keyID string, pluginConfig map[string]s
 	}, nil
 }
 
+func (s *pluginSigner) PluginAnnotations() map[string]string {
```
Comment (patrickzheng200) [inline signer/plugin.go:48]: We need a func doc for this function and mention that `pluginSigner implements notation.signerAnnotation`.

**Review by patrickzheng200 — COMMENTED** [review]
Looks like need to fix the unit tests, otherwise, LGTM.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Pull request merged**